### DISCUSSION TITLE
Dúvida sobre conceito de injeção de dependência e MVC e Minimal

### DISCUSSION THREAD

**Discussion by oimarcella** [balta-io/forum#645] [Tira Dúvidas] [answered]
Tópico 1:

Vi as aulas no curso de ASP NET 6, porém ficou muito nublado para mim quando eu sei que preciso lidar com a injeção de dependência.
No caso das aulas ele fez somente para o TokenService, só que eu estou pesquisando e lendo materiais extras  e vi um caso de um CustomerRepository sendo injetado, até usando o AddTransient.

E eu fiquei bem confusa sobre como eu deveria pensar para reconhecer que preciso fazer injeção de dependência no meu projeto.

Tópico 2:
Eu assisti as aulas do curso de ASP NET 6 e vi minimal primeiro, e em seguida MVC, fiquei imaginando que o segundo era uma continuação e aprofundamento de outro. Só que recentemente entrei num projeto que usa razor pages e minimal não MVC, então fiquei bem confusa se isso são 2 abordagens diferentes para um mesmo objetivo, ou são complementos.
![image](https://github.com/user-attachments/assets/90584cde-00ca-41b7-99b1-39878182d552)

**Comment by henriqueweege** [comment]
Oi, @oimarcella ! Tudo certo?

Tópico 1:

Sempre que houver dependência de outras classes é importante usar injeção de dependência. Quando dependemos de uma interface ao invés de um tipo concreto, conseguimos desacoplar nosso módulo, e usando injeção de dependência podemos mudar a implementação dessa interface sem quebrar partes que dependem dessa interface. Vale a pena ler sobre o último princípio do SOLID, Dependency Inversion.

Tópico 2:

MVC é um padrão de arquitetura. Minimal API é uma maneira de criar API's web. Se não estou enganado, você pode ter o padrão MVC utilizando Minimal API, porém, nem sempre, depende bastante do que o projeto precisa. A abordagem "contrária" ao Minimal API são as API's baseadas em controllers. A principal diferença é que ao invés de toda config de rota/retorno ser feita no Program, você tem classes que são responsáveis por isso.
Eu não trabalhei muito com Minimal API, então se alguém puder complementar em algum ponto, fique a vontade!

Valeeeu!

**Reply by oimarcella** [reply]
Agradeço pela resposta!

Então, mas rotas baseadas em controllers também tem sua organização separada em classes, não?
Acho que não entendi oque quis dizer com toda configuração ficar no program.

**Reply by henriqueweege** [reply, marked as answer]
@oimarcella, 
Sim, tem! Quando construímos uma Web API baseada em controllers, existem classes que tem a responsabilidade de serem controllers, e as rotas podem ser construídas com base nessa abordagem. Na parte "MVC" o Balta fala sobre isso.
A aula "Mapeando uma requisição" da parte "Introdução e Minimal API" deve esclarecer o segundo ponto.
